terraform/workspaces/infra/s3/s3.tf

```diff
@@ -185,6 +133,158 @@ resource "aws_iam_user_policy" "app" {
 EOF
 }
 
+###########################################
+# CDN bucket
+###########################################
+
+resource "aws_s3_bucket" "cdn" {
+  bucket        = "${var.workspace}-cdn"
+  force_destroy = var.force_destroy
+
+  logging {
+    target_bucket = var.cloudtrail_s3_bucket
+    target_prefix = "${var.workspace}-cdn/"
+  }
+  cors_rule {
+    allowed_headers = ["*"]
+    allowed_methods = ["HEAD", "POST"]
+    allowed_origins = ["*"]
+    expose_headers  = ["ETag"]
+    max_age_seconds = 3000
+  }
+}
+
+resource "aws_s3_bucket_versioning" "cdn" {
+  bucket = aws_s3_bucket.cdn.id
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+resource "aws_s3_bucket_server_side_encryption_configuration" "cdn" {
+  bucket = aws_s3_bucket.cdn.id
+  rule {
+    apply_server_side_encryption_by_default {
+      sse_algorithm = "AES256"
+    }
+  }
+}
+
+resource "aws_s3_bucket_ownership_controls" "cdn" {
+  bucket = aws_s3_bucket.cdn.id
+  rule {
+    object_ownership = "ObjectWriter"
+  }
+}
+
+resource "aws_s3_bucket_public_access_block" "cdn" {
+  bucket = aws_s3_bucket.cdn.id
+
+  block_public_acls       = false
+  block_public_policy     = false
+  ignore_public_acls      = false
+  restrict_public_buckets = false
+}
+
+resource "aws_s3_bucket_acl" "cdn" {
+  depends_on = [
+    aws_s3_bucket_ownership_controls.cdn,
+    aws_s3_bucket_public_access_block.cdn,
+  ]
+
+  bucket = aws_s3_bucket.cdn.id
+  acl    = "public-read"
+}
+
+data "aws_iam_policy_document" "cdn_bucket_policy" {
+  statement {
+    sid = "AllowAnonymousReads"
+    actions = [
+      "s3:GetObject",
+      "s3:GetObjectVersion"
+    ]
+    effect    = "Allow"
+    resources = ["${aws_s3_bucket.cdn.arn}/*"]
+
+    principals {
+      type        = "*"
+      identifiers = ["*"]
+    }
+  }
+
+  # NOTE: Insecure (non-SSL) requests are allowed for this bucket otherwise requests from Minio fail
+}
+
+resource "aws_s3_bucket_policy" "cdn_bucket" {
+  depends_on = [
+    aws_s3_bucket_acl.cdn
+  ]
+
+  bucket = aws_s3_bucket.cdn.id
+  policy = data.aws_iam_policy_document.cdn_bucket_policy.json
+}
+
+###########################################
+# Logs bucket
+###########################################
+
+resource "aws_s3_bucket" "logs" {
+  count = var.disable_logs ? 0 : 1
+
+  bucket        = "${var.workspace}-logs"
+  force_destroy = var.force_destroy
+}
+
+resource "aws_s3_bucket_ownership_controls" "logs" {
+  count = var.disable_logs ? 0 : 1
+  bucket = aws_s3_bucket.logs[count.index].id
+
+  rule {
+    object_ownership = "BucketOwnerEnforced"
+  }
+}
+
+resource "aws_s3_bucket_server_side_encryption_configuration" "logs" {
+  count  = var.disable_logs ? 0 : 1
+  bucket = aws_s3_bucket.logs[count.index].id
+  rule {
+    apply_server_side_encryption_by_default {
+      sse_algorithm = "AES256"
+    }
+  }
+}
+
```

Comment:
I think this also needs `aws_s3_bucket_public_access_block`

Reply:
`cdn` is the only resource with `aws_s3_bucket_public_access_block` because it's the only one that needs to be publicly available. By default, an s3 bucket wouldn't be publically accessible and we don't want `logs` to be. Is there another reason you would like to see `aws_s3_bucket_public_access_block`?